src/rpc.rs

```diff
@@ -31,12 +28,36 @@ pub enum Request {
 impl Request {
     /// Read a request from the given readable.
     pub fn read(conn: &mut impl Read, expected_chain_id: &chain::Id) -> Result<Self, Error> {
-        let msg_bytes = read_msg(conn)?;
-
-        // Parse Protobuf-encoded request message
-        let msg = proto::privval::Message::decode_length_delimited(msg_bytes.as_ref())
-            .map_err(|e| format_err!(ErrorKind::ProtocolError, "malformed message packet: {}", e))?
-            .sum;
+        let mut msg_bytes: Vec<u8> = vec![];
+        let msg;
+
+        // fix for Sei: collect incoming bytes of Protobuf from incoming msg
+        loop {
```

Comment:
Wonder if this logic should go to `tendermint_proto`, e.g. `proto::privval::Message::decode()`

Reply:
maybe not to proto, since proto shouldn't care about transport. Rather to `tendermint_p2p`, as tmkms author mentioned.